aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -218,13 +221,32 @@ def __init__(
         cors: bool,
         compress: bool,
         cache_control: Optional[str],
+        middleware: Optional[List[Callable]],
     ):
         self.method = method.upper()
         self.rule = rule
         self.func = func
         self.cors = cors
         self.compress = compress
         self.cache_control = cache_control
+        self.middleware = middleware or []
+
+    def __call__(self, router_middleware: List[Callable], app, args: Dict[str, str]):
+        """Builds the middleware stack using global and route middlewares, redefining the original handler function"""
+        if not self._middleware_built:
+            # prepend global router middleware first
+            all_middleware = list(router_middleware) + list(self.middleware)
```

Comment:
```suggestion
            all_middlewares = router_middlewares + self.middlewares
```

Both are already lists, we can save additional CPU cycles and memory. If you're worried about `router_middlewares` not being a list, we can treat that earlier. If someone subclass and calls calls a `Route` without a list, it'll raise an exception (EAFP)